mas-foundation/src/main/java/com/ca/mas/core/oauth/OAuthTokenClient.java

```diff
@@ -176,16 +178,18 @@ public OAuthTokenResponse obtainTokenUsingRefreshToken(@NonNull String refreshTo
                 .build();
 
         OAuthTokenResponse tokenResponse = null;
-
-        //Remove refresh token once we get a proper response from the server, no matter success or not.
-        mssoContext.takeRefreshToken();
-
         try {
             tokenResponse = new OAuthTokenResponse(obtainServerResponseToPostedForm(tokenRequest));
+            if(tokenRequest != null){
```

Comment:
This check can be removed. tokenRequest is always not null here.

if you really want to enforce the check, think of moving the **mssoContext.takeRefreshToken()** line right after the **validate** statement.

Reply:
incorporated these changes.

Reply:
Looks like you misunderstood my comment. Probably my bad.
This if-condition always evaluates to TRUE. Remove it.

Reply:
Sorry. I understood differently. Anyway, I made the changes.